jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -522,21 +522,16 @@ public void configureTempDirectory(File dir, WebAppContext context)
         if (dir == null)
             throw new IllegalArgumentException("Null temp dir");
 
-        //if dir exists and we don't want it persisted, delete it
-        if (dir.exists() && !context.isPersistTempDirectory())
+        // if it doesn't exist make it
```

Comment:
I'm not sure about this .... we need to check that if the dir should not be persisted then anything that is already in it should be removed before we try and unpack the webapp into it.

Reply:
But we cannot remove it arbitrarily here, only under small circumstance will removing the directory itself be safe.

This step isn't really necessary as both the WebInfConfiguration.deconfigure() and the Temp directory specific File.deleteOnExit() both cover the deletion angle.

As an alternative, we could simply clear out the directory contents.

Reply:
yep I think we need to clear it out

Reply:
Yes, I'd prefer to clear out the directory contents, and leave the dir in place.

This is safeguarding if the previous deconfigure() failed, and deleteOnExit might not come into play because you're just restarting the context/server without restarting the jvm.

Reply:
The temp directory deletion block has been restored.

Reply:
In the `Files.createTempDirectory` path (or the older `File.createTempFile` path), this delete would never have deleted the prior directory, as each time you enter this block you have a new directory path.